src/Domain/CoordinateMaps/TimeDependent/Shape.hpp

```diff
@@ -241,6 +243,16 @@ class Shape {
       const std::array<T, 3>& source_coords, double time,
       const FunctionsOfTimeMap& functions_of_time) const;
 
+  /*!
+   * \brief An optimized call that computes the target coordinates, frame
+   * velocity and jacobian at once to avoid duplicate calculations.
+   */
+  void coords_frame_velocity_jacobian(
+      gsl::not_null<std::array<DataVector, 3>*> source_and_target_coords,
```

Comment:
Sorry one last thing since I know I'll get confused in the future. Can you add `\details` about how the source coords are expected to be passed in to `source_and_target_coords`, but the resulting mapped coords are then also placed in `source_and_target_coords` at the end of the function once the source coords have been used.